### PR TITLE
feature/user-prferences-frontend

### PR DESCRIPTION
**Pull request by IrynaRobulets**
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/105%23discussion_r61234204%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/105%23discussion_r61235083%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/105%23discussion_r61235158%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/105%23discussion_r61235223%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/105%23discussion_r61235288%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/105%23discussion_r61239352%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/105%23discussion_r61242858%22%5D%2C%20%22comments%22%3A%20%7B%22Pull%20cdc0427e070dd831ea7b1fec29ef785a2ba47bd5%20DevelopGitHub/GitHubExtension/GitHubExtension.EntryPoint/FrontEnd/src/client/app/core/constants.js%207%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/105%23discussion_r61242858%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22again%3F%20%3A%29%22%2C%20%22created_at%22%3A%20%222016-04-27T11%3A37%3A05Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7273003%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20DevelopGitHub/GitHubExtension/GitHubExtension.EntryPoint/FrontEnd/src/client/app/core/constants.js%3AL16-24%22%7D%2C%20%22Pull%206b16de3944987742f211693a54981e576057e65d%20DevelopGitHub/GitHubExtension/GitHubExtension.EntryPoint/FrontEnd/src/client/app/core/constants.js%206%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/105%23discussion_r61235158%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22just%20%60USER_PHOTO%60%20or%20smt%20like%20that%22%2C%20%22created_at%22%3A%20%222016-04-27T10%3A25%3A17Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7273003%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20DevelopGitHub/GitHubExtension/GitHubExtension.EntryPoint/FrontEnd/src/client/app/core/constants.js%3AL16-24%22%7D%2C%20%22Pull%206b16de3944987742f211693a54981e576057e65d%20DevelopGitHub/GitHubExtension/GitHubExtension.EntryPoint/FrontEnd/src/client/app/core/constants.js%207%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/105%23discussion_r61234204%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22unnecessary%20%3A%29%22%2C%20%22created_at%22%3A%20%222016-04-27T10%3A16%3A56Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7273003%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20DevelopGitHub/GitHubExtension/GitHubExtension.EntryPoint/FrontEnd/src/client/app/core/constants.js%3AL16-24%22%7D%2C%20%22Pull%206b16de3944987742f211693a54981e576057e65d%20DevelopGitHub/GitHubExtension/GitHubExtension.EntryPoint/FrontEnd/src/client/app/userpreferences/userpreferences.controller.js%2014%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/105%23discussion_r61235083%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%222%20questions.%5Cr%5Cn-%20URL%20is%20called%20%60API_URL.UPLOADPHOTO%60%20and%20function%20is%20%60uploadPhoto%60%20but%20actually%20we%20are%20performing%20GET%20request.%20Why%3F%5Cr%5Cn-%20controller%20should%20not%20be%20aware%20of%20the%20API%20routes.%20so%20API_URL%20should%20be%20included%20to%20service%20%28not%20controller%29%22%2C%20%22created_at%22%3A%20%222016-04-27T10%3A24%3A31Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7273003%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20DevelopGitHub/GitHubExtension/GitHubExtension.EntryPoint/FrontEnd/src/client/app/userpreferences/userpreferences.controller.js%3AL9-35%22%7D%2C%20%22Pull%206b16de3944987742f211693a54981e576057e65d%20DevelopGitHub/GitHubExtension/GitHubExtension.EntryPoint/FrontEnd/src/client/app/userpreferences/userpreferences.controller.js%2023%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/105%23discussion_r61235223%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22remove%20console%20log%2C%20add%20comment%20why%20do%20we%20need%20this%20dirty%20hack%20%3A%29%22%2C%20%22created_at%22%3A%20%222016-04-27T10%3A25%3A56Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7273003%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20DevelopGitHub/GitHubExtension/GitHubExtension.EntryPoint/FrontEnd/src/client/app/userpreferences/userpreferences.controller.js%3AL9-35%22%7D%2C%20%22Pull%206b16de3944987742f211693a54981e576057e65d%20DevelopGitHub/GitHubExtension/GitHubExtension.EntryPoint/FrontEnd/src/client/app/userpreferences/userpreferences.route.js%205%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/105%23discussion_r61235288%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22does%20%60fa-tachometer%60%20work%3F%20%3A%29%22%2C%20%22created_at%22%3A%20%222016-04-27T10%3A26%3A30Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7273003%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF%22%7D%7D%2C%20%7B%22body%22%3A%20%22Nope%20%3A%29%22%2C%20%22created_at%22%3A%20%222016-04-27T11%3A01%3A02Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/17322068%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/IrynaRobulets%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20DevelopGitHub/GitHubExtension/GitHubExtension.EntryPoint/FrontEnd/src/client/app/userpreferences/userpreferences.route.js%3AL23-30%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>
- [ ] <a href='#crh-comment-Pull 6b16de3944987742f211693a54981e576057e65d DevelopGitHub/GitHubExtension/GitHubExtension.EntryPoint/FrontEnd/src/client/app/core/constants.js 7'></a> &nbsp;<b><a href='https://github.com/ITsvetkoFF/Kv-013/pull/105#discussion_r61234204'>File: DevelopGitHub/GitHubExtension/GitHubExtension.EntryPoint/FrontEnd/src/client/app/core/constants.js:L16-24</a></b>
- <a href='https://github.com/ITsvetkoFF'></a> unnecessary :)
- [x] <a href='#crh-comment-Pull 6b16de3944987742f211693a54981e576057e65d DevelopGitHub/GitHubExtension/GitHubExtension.EntryPoint/FrontEnd/src/client/app/userpreferences/userpreferences.controller.js 14'></a> &nbsp;<b><a href='https://github.com/ITsvetkoFF/Kv-013/pull/105#discussion_r61235083'>File: DevelopGitHub/GitHubExtension/GitHubExtension.EntryPoint/FrontEnd/src/client/app/userpreferences/userpreferences.controller.js:L9-35</a></b>
- <a href='https://github.com/ITsvetkoFF'></a> 2 questions.
- URL is called `API_URL.UPLOADPHOTO` and function is `uploadPhoto` but actually we are performing GET request. Why?
- controller should not be aware of the API routes. so API_URL should be included to service (not controller)
- [x] <a href='#crh-comment-Pull 6b16de3944987742f211693a54981e576057e65d DevelopGitHub/GitHubExtension/GitHubExtension.EntryPoint/FrontEnd/src/client/app/core/constants.js 6'></a> &nbsp;<b><a href='https://github.com/ITsvetkoFF/Kv-013/pull/105#discussion_r61235158'>File: DevelopGitHub/GitHubExtension/GitHubExtension.EntryPoint/FrontEnd/src/client/app/core/constants.js:L16-24</a></b>
- <a href='https://github.com/ITsvetkoFF'></a> just `USER_PHOTO` or smt like that
- [x] <a href='#crh-comment-Pull 6b16de3944987742f211693a54981e576057e65d DevelopGitHub/GitHubExtension/GitHubExtension.EntryPoint/FrontEnd/src/client/app/userpreferences/userpreferences.controller.js 23'></a> &nbsp;<b><a href='https://github.com/ITsvetkoFF/Kv-013/pull/105#discussion_r61235223'>File: DevelopGitHub/GitHubExtension/GitHubExtension.EntryPoint/FrontEnd/src/client/app/userpreferences/userpreferences.controller.js:L9-35</a></b>
- <a href='https://github.com/ITsvetkoFF'></a> remove console log, add comment why do we need this dirty hack :)
- [ ] <a href='#crh-comment-Pull 6b16de3944987742f211693a54981e576057e65d DevelopGitHub/GitHubExtension/GitHubExtension.EntryPoint/FrontEnd/src/client/app/userpreferences/userpreferences.route.js 5'></a> &nbsp;<b><a href='https://github.com/ITsvetkoFF/Kv-013/pull/105#discussion_r61235288'>File: DevelopGitHub/GitHubExtension/GitHubExtension.EntryPoint/FrontEnd/src/client/app/userpreferences/userpreferences.route.js:L23-30</a></b>
- <a href='https://github.com/ITsvetkoFF'></a> does `fa-tachometer` work? :)
- <a href='https://github.com/IrynaRobulets'></a> Nope :)
- [ ] <a href='#crh-comment-Pull cdc0427e070dd831ea7b1fec29ef785a2ba47bd5 DevelopGitHub/GitHubExtension/GitHubExtension.EntryPoint/FrontEnd/src/client/app/core/constants.js 7'></a> &nbsp;<b><a href='https://github.com/ITsvetkoFF/Kv-013/pull/105#discussion_r61242858'>File: DevelopGitHub/GitHubExtension/GitHubExtension.EntryPoint/FrontEnd/src/client/app/core/constants.js:L16-24</a></b>
- <a href='https://github.com/ITsvetkoFF'></a> again? :)

<a href='https://www.codereviewhub.com/ITsvetkoFF/Kv-013/pull/105?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/ITsvetkoFF/Kv-013/pull/105?approve=1'></a>&nbsp;<a href='https://github.com/ITsvetkoFF/Kv-013/pull/105'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by ITsvetkoFF** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/ITsvetkoFF/Kv-013/pull/105'>Refresh</a>.<a href='#crh-update-none'></a>

File: DevelopGitHub/GitHubExtension/GitHubExtension.EntryPoint/FrontEnd/src/client/app/core/constants.js [hunk 1]
```diff
@@ -16,8 +16,8 @@
                 LOGOUT: resource + 'account/logout',
                 internalActivityUrl: resource + 'activity/internal',
                 externalActivityUrl: resource + 'activity/external?page=',
-                UPLOADPHOTO: resource + 'account/avatar',
-                REPOSITORY: resource + 'user/repos',
+                UPLOADPHOTO: resource + 'user/avatar',
+				REPOSITORY: resource + 'user/repos',
```
Comment (ITsvetkoFF) [inline DevelopGitHub/GitHubExtension/GitHubExtension.EntryPoint/FrontEnd/src/client/app/core/constants.js]: unnecessary :)

File: DevelopGitHub/GitHubExtension/GitHubExtension.EntryPoint/FrontEnd/src/client/app/userpreferences/userpreferences.controller.js [hunk 2]
```diff
@@ -9,15 +9,26 @@
...
 
         activate();
 
         function activate() {
+            uploadPhoto();
             logger.info('Activate User Preferences View');
         }
 
+        function uploadPhoto() {
+            return userData.getImage(API_URL.UPLOADPHOTO).then(function (newImageUrl) {
```
Comment (ITsvetkoFF) [inline DevelopGitHub/GitHubExtension/GitHubExtension.EntryPoint/FrontEnd/src/client/app/userpreferences/userpreferences.controller.js]: 2 questions.
- URL is called `API_URL.UPLOADPHOTO` and function is `uploadPhoto` but actually we are performing GET request. Why?
- controller should not be aware of the API routes. so API_URL should be included to service (not controller)

File: DevelopGitHub/GitHubExtension/GitHubExtension.EntryPoint/FrontEnd/src/client/app/core/constants.js [hunk 3]
```diff
@@ -16,8 +16,8 @@
                 LOGOUT: resource + 'account/logout',
                 internalActivityUrl: resource + 'activity/internal',
                 externalActivityUrl: resource + 'activity/external?page=',
-                UPLOADPHOTO: resource + 'account/avatar',
-                REPOSITORY: resource + 'user/repos',
+                UPLOADPHOTO: resource + 'user/avatar',
```
Comment (ITsvetkoFF) [inline DevelopGitHub/GitHubExtension/GitHubExtension.EntryPoint/FrontEnd/src/client/app/core/constants.js]: just `USER_PHOTO` or smt like that

File: DevelopGitHub/GitHubExtension/GitHubExtension.EntryPoint/FrontEnd/src/client/app/userpreferences/userpreferences.controller.js [hunk 4]
```diff
@@ -9,15 +9,26 @@
...
+                vm.imageSource = newImageUrl;
+            });
+        }
+
         vm.uploadFile = function (event) {
-            userData.postImage(API_URL.UPLOADPHOTO, event.target.files);
+            return userData.postImage(API_URL.UPLOADPHOTO, event.target.files).then(function (newImageUrl) {
+                console.log(newImageUrl);
+                vm.imageSource = newImageUrl +'?r='+ Math.round(Math.random() * 999999);
```
Comment (ITsvetkoFF) [inline DevelopGitHub/GitHubExtension/GitHubExtension.EntryPoint/FrontEnd/src/client/app/userpreferences/userpreferences.controller.js]: remove console log, add comment why do we need this dirty hack :)

File: DevelopGitHub/GitHubExtension/GitHubExtension.EntryPoint/FrontEnd/src/client/app/userpreferences/userpreferences.route.js [hunk 5]
```diff
@@ -23,7 +23,7 @@
                     title: 'Preferences',
                     settings: {
                         nav: 6,
-                        content: '<i class="fa fa-dashboard"></i>' + i18n.message.PREFERENCES
+                        content: '<i class="fa fa-tachometer"></i>' + i18n.message.PREFERENCES
```
Comment (ITsvetkoFF) [inline DevelopGitHub/GitHubExtension/GitHubExtension.EntryPoint/FrontEnd/src/client/app/userpreferences/userpreferences.route.js]: does `fa-tachometer` work? :)

File: DevelopGitHub/GitHubExtension/GitHubExtension.EntryPoint/FrontEnd/src/client/app/userpreferences/userpreferences.controller.js (hunk 2 above)
Comment (IrynaRobulets) [inline DevelopGitHub/GitHubExtension/GitHubExtension.EntryPoint/FrontEnd/src/client/app/userpreferences/userpreferences.controller.js]: I have the same url for post and get request. And both requests return image url.

File: DevelopGitHub/GitHubExtension/GitHubExtension.EntryPoint/FrontEnd/src/client/app/userpreferences/userpreferences.route.js (hunk 5 above)
Comment (IrynaRobulets) [inline DevelopGitHub/GitHubExtension/GitHubExtension.EntryPoint/FrontEnd/src/client/app/userpreferences/userpreferences.route.js]: Nope :)

File: DevelopGitHub/GitHubExtension/GitHubExtension.EntryPoint/FrontEnd/src/client/app/core/constants.js [hunk 6]
```diff
@@ -16,8 +16,8 @@
                 LOGOUT: resource + 'account/logout',
                 internalActivityUrl: resource + 'activity/internal',
                 externalActivityUrl: resource + 'activity/external?page=',
-                UPLOADPHOTO: resource + 'account/avatar',
-                REPOSITORY: resource + 'user/repos',
+                USER_PHOTO: resource + 'user/avatar',
+				REPOSITORY: resource + 'user/repos',
```
Comment (ITsvetkoFF) [inline DevelopGitHub/GitHubExtension/GitHubExtension.EntryPoint/FrontEnd/src/client/app/core/constants.js]: again? :)

**Pull request merged**